pkg/reconciler/instances/istio/reset/pod/reset/action_test.go

```diff
@@ -145,4 +148,142 @@ func Test_DefaultPodsResetAction_Reset(t *testing.T) {
+			action := NewDefaultPodsResetAction(pod.NewParentKindMatcher())
+			pl := v1.PodList{
+				Items: []v1.Pod{pendingPod},
+			}
+			fixWaitOpts.Interval = time.Millisecond * 10
+			fixWaitOpts.Timeout = time.Millisecond * 100
+
+			// when
+			err := action.Reset(context.Background(), k8sclient, fixRetryOpts, pl, log, false, fixWaitOpts)
+			require.Error(t, err)
+
+			// then
+			got, err := k8sclient.CoreV1().Pods(podNs).List(ctx, metav1.ListOptions{
+				LabelSelector: labelSelector,
+			})
+			require.NoError(t, err)
+			for _, p := range got.Items {
+				v, ok := p.Annotations[pod.AnnotationResetWarningKey]
+				require.True(t, ok)
+				require.Equal(t, pod.AnnotationResetWarningRolloutTimeoutVal, v)
+			}
+		})
+	})
+
+	t.Run("given pod without owner ref, should annotate", func(t *testing.T) {
+		// given
+		podNs := "testNamespace"
+		podName := "testPod"
+		pendingPod := v1.Pod{
+			TypeMeta: metav1.TypeMeta{APIVersion: "v1", Kind: "Pod"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: podNs,
+			},
+		}
+		k8sclient := fake.NewSimpleClientset(&pendingPod)
+		action := NewDefaultPodsResetAction(pod.NewParentKindMatcher())
+		pl := v1.PodList{
+			Items: []v1.Pod{pendingPod},
+		}
+
+		// when
+		err := action.Reset(context.Background(), k8sclient, fixRetryOpts, pl, log, false, fixWaitOpts)
+		require.NoError(t, err)
+
+		// then
+		got, err := k8sclient.CoreV1().Pods(podNs).Get(context.Background(), podName, metav1.GetOptions{})
+		require.NoError(t, err)
+		v, ok := got.Annotations[pod.AnnotationResetWarningKey]
+		require.True(t, ok)
+		require.Equal(t, pod.AnnotationResetWarningNoOwnerVal, v)
+	})
+
+	t.Run("given pods in not running state but they have been already annotated, no action should be performed", func(t *testing.T) {
+		// given
+		matcher := mocks.Matcher{}
+		action := NewDefaultPodsResetAction(&matcher)
+		simplePodAnnotated := simplePod
+		if simplePodAnnotated.Annotations == nil {
+			simplePodAnnotated.Annotations = make(map[string]string)
+		}
+		simplePodAnnotated.Annotations[pod.AnnotationResetWarningKey] = "test"
+		pods := []v1.Pod{simplePodAnnotated, simplePod}
+
+		handlersMap := map[pod.Handler][]pod.CustomObject{}
+
+		matcher.On("GetHandlersMap", mock.Anything, mock.AnythingOfType("[]retry.Option"), mock.AnythingOfType("v1.PodList"), mock.AnythingOfType("*zap.SugaredLogger"), mock.AnythingOfType("bool"), mock.AnythingOfType("pod.WaitOptions")).
+			Return(handlersMap)
+
+		// when
+		err := action.Reset(ctx, kubeClient, fixRetryOpts, v1.PodList{Items: pods}, log, debug, fixWaitOpts)
+
+		// then
+		require.NoError(t, err)
+		shouldBeCalledWithoutAnnotatePod := func(t testing.TB) {
+			t.Helper()
+			matcher.AssertCalled(t, "GetHandlersMap", kubeClient, fixRetryOpts, v1.PodList{Items: []v1.Pod{simplePod}}, log, debug, fixWaitOpts)
+		}
+		shouldBeCalledWithoutAnnotatePod(t)
+	})
+}
+
+func createLabelSelector(labels map[string]string) string {
```

Comment:
I kind of don't like the fact that you duplicate the logic used in the production code but I understand that it is for convenience reasons